src/main/java/seedu/address/model/person/Customer.java

```diff
@@ -26,17 +26,42 @@
      * @param address
      * @param tags
      */
-
     public Customer(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
         super(name, phone, email, address, tags);
         id = ++idCount;
     }
 
+    public Customer(Name name, Phone phone, Email email, Address address, Set<Tag> tags, String id) {
+        super(name, phone, email, address, tags);
+        this.id = Integer.parseInt(id);
```

Comment:
`parseInt` could throw an exception. Maybe let the parser handle `parseInt` and the exception there?

Reply:
I think hes's trying to create a constructor to be used in JsonAdaptedCustomer.

So i think what you can do is:
At JsonAdaptedCustomer,
-Check if id is null
-Check if the id is valid using a method Customer.isValidId()
-then parseInt the String id
-then create the new Customer(INT id, Name name....)

Refer to JsonAdaptedTask's task id

Reply:
removed